WordPress/gutenberg
# Add stylelint rule to prevent usage of flex-direction reverse values

Fixes https://github.com/WordPress/gutenberg/issues/63048

## What?

The `flex-direction` CSS property values `row-reverse` and `column-reverse` should not be used when they make visual order and DOM order mismatch in a way that impacts meaning and interaction. That is, in _almost all cases_ they should not be used. See the issue for more details.

## Why?

For keyboard users and assistive technology users, the visual order and DOM order must match.

## How?

- Adds a `declaration-property-value-allowed-list` rule for `flex-direction` allowing only the following values:
  - "row"
  - "column"
  - "inherit"
  - "initial"
  - "unset"
- The rule will trigger an error with a custom message.
- Refactors the implementation for the experimental `core/form-input` block. The checkbox label is no longer swapped via CSS. There's now a condition to render the checkbox and its label in the expected order. This required to modify the edit and save functions of the block. I'm not too concerned about it as this is still flagged as a Gutenberg experiment.
- Adds 7 disable comments for the rule as follows:

**Valid cases:**
- 2 occurrences in `packages/block-library/src/navigation/style.scss`
- 1 occurrence in `packages/block-library/src/page-list/style.scss`
These flex-direction properties use a CSS variable `--navigation-layout-direction` that would be flagged as an error. To my understanding, the variable value is either 'row', 'column' or 'initial' so it's OK.

**Cases that need refactoring. I will split them in separate issues:**
- 1 occurrence in `packages/block-editor/src/hooks/block-hooks.scss`
- 1 occurrence in `packages/block-editor/src/hooks/use-editor-wrapper-styles.native.scss`
- 2 occurrences in `packages/block-library/src/media-text/style.native.scss`

## Testing Instructions




- Run the CSS linter in your terminal: `npm run lint:css`.
- Observe there are no errors reported.
- Uncomment one of the disable comments added in this PR.
- Run the CSS linter again.
- Observe the reported error in your console.
- In the WP admin, go to Gutenberg > Experiments, and enable the `Form and input blocks` experiment.
- Add a form block to a post.
- Select the form and add a checkbox input.
- Observe the checkbox input comes first and its label after.
- Inspect the source and observe the DOM order matches the visual order.
- Save and go to the front end.
- Observe the checkbox and its label visual order matches the DOM order.


### Testing Instructions for Keyboard


## Screenshots or screencast
